### PR TITLE
test: migrate to ES6

### PR DESCRIPTION
**Pull request by SukkaW**
### `Notable Changes`

Similar to #336. This time test has been migrated to ES6.

#### `Commit Message Summary (CHANGELOG)`

```
test: migrate to ES6
```

### `Type`

> ℹ️  What types of changes does your code introduce?

> 👉 _Put an `x` in the boxes that apply and delete all others_

- [ ] CI
- [ ] Fix
- [ ] Perf
- [ ] Docs
- [x] Test
- [ ] Chore
- [ ] Style
- [ ] Build
- [ ] Feature
- [x] Refactor

### `SemVer`

> ℹ️  What changes to the current `semver` range does your PR introduce?

> 👉  _Put an `x` in the boxes that apply and delete all others_

- [ ] Fix (:label: Patch)
- [ ] Feature (:label: Minor)
- [ ] Breaking Change (:label: Major)

### `Issues`

> ℹ️ What issue(s) (if any) are closed by your PR?

> 👉  _Replace `#1` with the issue number that applies and removes the ``` ` ```_

- Fixes `#1`

### `Checklist`

> ℹ️  You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. This is a reminder of what we are going to look for before merging your code.

> 👉  _Put an `x` in the boxes that apply and delete all others._

- [x] Lint and unit tests pass with my changes
- [ ] I have added tests that prove my fix is effective/works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes are merged and published in downstream modules

### CODE REVIEW

File: test/parser.js [hunk 1]
```diff
@@ -1,20 +1,112 @@
...
+
+  it('directive ?php with multi html', done => {
+    expect(posthtml()
+      .process(multiHTML, options)
+      .then(result => {
+        expect(multiHTML).to.eql(result.html)
+        done()
+      })
+      .catch(error => done(error))
+    )
```
Comment (Scrum) [inline test/parser.js:110]: Something went wrong here

File: test/source.js [hunk 2]
```diff
@@ -1,36 +1,36 @@
...
-    .catch(function (error) {
+    .catch(error => {
       done(error)
     })
 }
 
-describe('Source', function () {
-  it('Source code must not mutate', function (done) {
+describe('Process', () => {
+  it('should not throw on empty tree', done => {
```
Comment (Scrum) [inline test/source.js:33]: Something went wrong here

**Review by Scrum — CHANGES_REQUESTED** [review]

File: test/parser.js (hunk 1 above)
Comment (SukkaW) [reply test/parser.js:110]: Yeah, you are right! Something went wrong with my own written script. I will rebase the PR.

File: test/source.js (hunk 2 above)
Comment (SukkaW) [reply test/source.js]: Rebased.

**Review by anikethsaha — APPROVED** [review]

**Review by Scrum — APPROVED** [review]

**Pull request merged**